tls/s2n_connection.c

```diff
@@ -267,6 +268,7 @@ static int s2n_connection_zero(struct s2n_connection *conn, int mode, struct s2n
     conn->verify_host_fn = NULL;
     conn->verify_host_fn_overridden = 0;
     conn->data_for_verify_host = NULL;
+    conn->preferred_key_shares = 0;
```

Comment:
Nit: technically this is unnecessary because of line 248, where we zero the memory of the connection structure.